omod/src/main/java/org/openmrs/module/openconceptlab/web/rest/resources/ImportResource.java

```diff
@@ -173,4 +180,25 @@ private UpdateScheduler getUpdateScheduler() {
     private static ImportService getImportService() {
         return Context.getService(ImportService.class);
     }
+
+    @Override
+    public Object upload(MultipartFile multipartFile, RequestContext requestContext) throws ResponseException, IOException {
+
+        File file = new File(multipartFile.getOriginalFilename());
+        multipartFile.transferTo(file);
+        ZipFile zipFile = new ZipFile(file);
+        InputStream inputStream = Utils.extractExportInputStreamFromZip(zipFile);
+        file.delete();
```

Comment:
Same issue here. You tell to delete the file yet you pass the inputStream of the deleted file to the importer. It doesn't seem right. It feels like the importer should handle deletion. To achieve that instead of passing inputStream you would have to pass the file, process it and delete at the end.